### PR TITLE
Remove unused API 

### PR DESCRIPTION
**Pull request by HubertJaworski**

### CODE REVIEW

File: neptune/backend.py [hunk 1]
```diff
@@ -144,10 +144,6 @@ def update_tags(self, experiment, tags_to_add, tags_to_delete):
     def upload_experiment_source(self, experiment, data, progress_indicator):
```
[the remark this answers is not in the document]
Comment (pkasprzyk) [reply neptune/backend.py:144]: `download_artifacts`'s only implementation also raises exception. Interestingly enough, it's used in `Experiment`

File: tests/neptune/internal/storage/test_datastream.py [hunk 2]
```diff
@@ -63,7 +63,9 @@ def test_generate_chunks_from_stream(self):
         text = u"ABCDEFGHIJKLMNOPRSTUWXYZ"
 
         # when
-        stream = FileChunkStream(UploadEntry(StringIO(text), "some/path"))
+        stream = FileChunkStream(
+            UploadEntry(BytesIO(bytes(text, "utf-8")), "some/path")
```
Comment (pkasprzyk) [inline tests/neptune/internal/storage/test_datastream.py:67]: this looks like a part of a different PR ;)

**Review by pkasprzyk — CHANGES_REQUESTED** [review]

File: tests/neptune/internal/storage/test_datastream.py (hunk 2 above)
Comment (HubertJaworski) [reply tests/neptune/internal/storage/test_datastream.py:67]: that is true.Will move it

File: neptune/backend.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (HubertJaworski) [reply neptune/backend.py:144]: This is not a complete cleanup of everything unused in that api - I didn't want to make a too deep of a purge to not extend the scope of this. This is a good idea to remove it in general, but I think that it should not be bundled in this MR

File: neptune/internal/api_clients/hosted_api_clients/hosted_alpha_leaderboard_api_client.py [hunk 3]
```diff
@@ -13,23 +13,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+
+# pylint: disable=too-many-lines
```
[the remark this answers is not in the document]
Comment (HubertJaworski) [reply neptune/internal/api_clients/hosted_api_clients/hosted_alpha_leaderboard_api_client.py:17]: tell this to me running pyling and it complaining

[the remark this answers is not in the document]
Comment (HubertJaworski) [reply neptune/internal/api_clients/hosted_api_clients/hosted_alpha_leaderboard_api_client.py:17]: maybe after I remove some functions I'll go lower than 1k lines

File: neptune/backend.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (HubertJaworski) [reply neptune/backend.py]: I removed all unused methods from backend.py

**Review by pkasprzyk — APPROVED** [review]

**Pull request merged**